solidity/contracts/igps/GasOverheadIgp.sol

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+// ============ Internal Imports ============
+import {IInterchainGasPaymaster} from "../../interfaces/IInterchainGasPaymaster.sol";
+// ============ External Imports ============
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+/**
+ * @notice An IGP that adds configured gas overheads to gas amounts and forwards
+ * calls to an "inner" IGP.
+ * @dev The intended use of this contract is to store overhead gas amounts for destination
+ * domains, e.g. Mailbox and/or ISM gas usage, such that users of this IGP are only required
+ * to specify the gas amount used by their own applications.
+ */
+contract GasOverheadIgp is IInterchainGasPaymaster, OwnableUpgradeable {
+    // ============ Public Storage ============
+
+    /// @notice The IGP that is called when paying for or quoting gas
+    /// after applying overhead gas amounts.
+    IInterchainGasPaymaster public innerIgp;
+
+    /// @notice Destination domain => overhead gas amount on that domain.
+    mapping(uint32 => uint256) public destinationGasOverhead;
```

Comment:
```suggestion
    bytes32 key = keccak256(destinationDomain, destinationAddressBytes32);
    mapping(bytes32 => uint256) public destinationGasOverhead;
```
do we need this for ISMs?

Reply:
Not quite sure what you mean, is `destinationAddressBytes32` the address of the ISM on the remote/destination chain? Imo this can just be implied -- like the particular GasOverheadIgp that's deployed will relate to e.g. the MultisigIsms on all remote chains, but it doesn't need to be aware of what the MultisigIsm addresses are